app/features/signup/verify-email.ts

```diff
@@ -1,40 +1,6 @@
-import { useEffect, useRef, useState } from 'react';
+import { useState } from 'react';
 import { useToast } from '~/hooks/use-toast';
-import {
-  useRequestNewEmailVerificationCode,
-  useVerifyEmail,
-} from '../user/user-hooks';
-
-export const useVerifyEmailOnLoad = ({
-  code,
-  onFailed,
-}: {
-  code: string | undefined;
-  onFailed: () => void;
-}) => {
-  const verifyEmail = useVerifyEmail();
-  const { toast } = useToast();
-  const failedRef = useRef(onFailed);
-
-  useEffect(() => {
```

Comment:
it's pretty cool that we can now get rid of useEffect for cases when we need to do something on initial load by just moving that to client loaders